vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -109,12 +108,17 @@ def pre_build(self):
             )
 
     def build(self):
-        dash_ag_grid_conf = {"id": self._callable_object_id} if hasattr(self, "_callable_object_id") else {}
+        # The pagination setting (and potentially others) only work when the initially built AgGrid has the same
+        # setting as the object that is built on-page-load and rendered finally.
+        dash_ag_grid_conf = self.figure._arguments.copy()
+        dash_ag_grid_conf["data_frame"] = pd.DataFrame()
+        if hasattr(self, "_callable_object_id"):
+            dash_ag_grid_conf["id"] = self._callable_object_id
```

Comment:
This is very clean solution! I just think that the following two lines:
```
        if hasattr(self, "_callable_object_id"):
            dash_ag_grid_conf["id"] = self._callable_object_id
```
 don't make any difference as the `id` is already copied to the `dash_ag_grid_conf` in the previous lines. 

Also, it seems like it makes sense to push similar approach for the vm.Table - dash_data_table solution.